pkg/querier/queryrange/results_cache.go

```diff
@@ -202,6 +223,34 @@ func (s resultsCache) handleMiss(ctx context.Context, r Request) (Response, []Ex
 		return response, []Extent{}, nil
 	}
 
+	cacheGenNumber := s.extractor.ExtractCacheGenNumber(response)
+
+	if cacheGenNumber == "-1" {
+		level.Debug(s.logger).Log("msg", "there is an inconsistency in cache generation numbers, not caching the response")
+		return response, []Extent{}, nil
+	}
+
+	if cacheGenNumber != "" {
+		if s.cacheGenNumbers[userID] == "" {
+			s.cacheGenNumbers[userID] = cacheGenNumber
+		} else {
+			// see if we have received higher gen number and start using it for caching new results using that
```

Comment:
Why are we doing this? Lets avoid this optimisation and let the query-frontend depend only on the periodic reload?